### ISSUE TITLE
GotoDefinition feature show incorrect result

### ISSUE THREAD

**Issue by herunlei** [microsoft/TypeScript#54874]
# Bug Report

### 🔎 Search Terms

`getDefinitionAtPosition`, `symbol.declarations`, `addDeclarationToSymbol`

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about declaration merging

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgGIHt3IN4FgBQyyowYAXMgBQCUyAvAHzIBu6wAJgNwEC+BBMAK4gEYYOhBpMNHH3wEEAGzgBnFVKx5CyAA5RgzOJGQAPesgCM3bQgkqwUQaPRQZ2ZGAAWwFQDozdMgATMhyBESQ9jQUrBw44UQenlDoAO7IIBDpAKJQKa4A5ACyEF7o7BnoYMQAtjqKEDUQ4BDsvgXU1kRyYfiZ6RjoNNZAA)

### 💻 Code

```ts
interface Foo {
  init: () => void;
}

function Foo() {}

class Foo {
  private x = 1;
  constructor() { this.x = 2 } // triggering gotoDefinition at 'this'

  test(): void {
    throw new Error('Method not implemented.');
  }
}

new Foo();
```

### 🙁 Actual behavior

The gotoDefinition returns 3 result including `interface Foo`, `class Foo`, and `function Foo`.

### 🙂 Expected behavior

Only return `Class Foo`.

**Comment by nmain** [comment]
This is invalid code; the existence of both `function Foo` and `class Foo` will fail at runtime.  Typescript does show this error.  Given that it's invalid, the behavior of listing both on go to definition is reasonable.  Until you remove or rename one to fix the problem, who can really be sure what you meant?

**Comment by fatcerberus** [comment]
For the record, including the interface in GTD is by design: it merges with the class, so in practice `Foo`‘s definition is split between both sites.
